src/Microsoft.Azure.WebJobs.Logging/Internal/TimeBucket.cs

```diff
@@ -27,5 +29,63 @@ public static long ConvertToBucket(DateTime dt)
             }
             return min;
         }
+
+        // Used to compute epoch. 
+        const long ticksBaseline = 633663648000000000L;
+        const long ticksPerMonth = 26785144663987L;
+        
+        public static DateTime CommonEpoch = new DateTime(0, DateTimeKind.Utc); // maps to epoch 0 
+        public static string CommonEpochSuffix = "common";
+
+        public static long GetEpochNumberFromTable(CloudTable table)
+        {
+            string tableName = table.Name;
+            if (tableName.Length < 5)
+            {
+                return -1;
+            }
+            string suffix = tableName.Substring(tableName.Length - 5, 5);
+            long epoch;
+            if (long.TryParse(suffix, out epoch))
+            {
+                return epoch;
+            }
+            return -1;
+        }
+
+        // Epoch must be positive, orderd integers.
+        // Use YYYYMM 
+        private static long GetEpochNumber(DateTime epoch)
```

Comment:
Per your comment below, we could indeed push some of this into the TableProvider interface.   But here are some forces we still need to account for: 
1. We need to be able to convert from Date --> Table and Table --> Date. 
2. Also, tables may not be consecutive (hence the need for ListTables).  We may have data for Jan 1st, nothing for 2 months, and then April 1st.  
3. minor issue: Table names are a very restricted character set (basically alpah numeric) and can't represent datetime.ToString(). so using integer also solved this problem. 
4. minor issue: I wanted the tables sorted lexographically in table viewing tools. DateTime doesn't sort like that. 



---
In reply to: [82908776](https://github.com/Azure/azure-webjobs-sdk/pull/862#discussion_r82908776) [](ancestors = 82908776)